### PR TITLE
chore: add tests

### PR DESCRIPTION
**Pull request by swissarmytowel**
## Summary

Adding tests and increasing overall coverage

## Checklist

Ensure you completed **all of the steps** below before submitting your pull request:

- [x] Add [natspec](https://docs.soliditylang.org/en/latest/natspec-format.html) for all functions / parameters
- [x] Ran `forge snapshot`
- [x] Ran `forge fmt`
- [x] Ran `forge test`
- [x] [Triage Slither issues](../README.md#triage-issues), and post uncertain ones in the PR

### CODE REVIEW

File: test/integration-tests/PomaceSettlement.t.sol [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        assertEq(sdyc.balanceOf(address(engine)), expectedEngineSdycBalance);
+
+        assertEq(weth.balanceOf(address(this)), 0);
+        assertEq(weth.balanceOf(address(engine)), engineWethBefore + 1 * 1e18);
+
+        assertEq(usdc.balanceOf(address(this)), selfUsdcBefore);
+        assertEq(usdc.balanceOf(address(engine)), engineUsdcBefore);
+    }
+
+    function testSettleCallWithNonUnderlyingNorStrike() public {
```
Comment (dsshap) [inline test/integration-tests/PomaceSettlement.t.sol:150]: this and `testSettlePutWithNonUnderlyingNorStrike` should be in their own test contract

File: test/libraries/BytesArrayUtil.t.sol [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import {BytesArrayUtil} from "../../src/libraries/BytesArrayUtil.sol";
+
+import "../../src/config/constants.sol";
+import "../../src/config/errors.sol";
+import "../../src/config/types.sol";
+
+/**
+ * Basic tests
+ */
+contract BytesArrayUtilTest is Test {}
```
Comment (dsshap) [inline test/libraries/BytesArrayUtil.t.sol:14]: Are there supposed to be tests here?

Comment (swissarmytowel) [reply test/libraries/BytesArrayUtil.t.sol]: Added in tests for `append` and `indexOf`

File: test/integration-tests/PomaceSettlement.t.sol (hunk 1 above)
Comment (dsshap) [reply test/integration-tests/PomaceSettlement.t.sol:185]: i meant create a new test contract with shared setup, some similarities between these two tests.

**Pull request merged**